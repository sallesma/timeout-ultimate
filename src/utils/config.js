export const Levels = {
  EASY: 'EASY',
  MIDDLE: 'MIDDLE',
  DIFFICULT: 'DIFFICULT',
  ANY: 'ANY',
};

export const Categories = {
  SPIRIT: 'Esprit du jeu',
  FIELD: 'Aire de jeu',
  TEAMS: 'Les Équipes',
  PULL: "L'Engagement",
  STATUS: 'Le statut du disque',
  COUNT: 'Le Compte',
  CHECK: 'Le Check',
  OUT: 'Hors des limites',
  RECEIVER: 'Réceptionneur et positionnement',
  TURNOVERS: 'Turnovers',
  GOAL: 'Marquer un point',
  FOULS_INFRACTIONS_VIOLATIONS: 'Fautes, infractions, violations',
  CONTINUATION: 'Continuité après un appel de faute ou violation',
  FOULS: 'Fautes',
  INFRACTIONS_VIOLATIONS: 'Infractions et violations',
  STOPPAGE: 'Stoppage',
  TIMEOUT: 'Temps-morts',
  APPENDIX: 'Appendice',
  HAND_SIGNALS: 'Gestes',
  INDOOR: 'Indoor',
};
