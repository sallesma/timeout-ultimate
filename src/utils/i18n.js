import { I18n } from 'i18n-js';
import * as Localization from 'expo-localization';

import en from './locales/en';
import fr from './locales/fr';
import AsyncStorage from '@react-native-async-storage/async-storage';

const i18n = new I18n({
  en,
  fr,
});

i18n.defaultLocale = 'fr';
i18n.fallbacks = true;
i18n.enableFallback = true;

AsyncStorage.getItem('language').then((lang) => {
  const allowedLang = ['fr', 'en'];
  if (!allowedLang.includes(lang)) {
    if (allowedLang.includes(Localization.locale.slice(0, 2))) i18n.locale = Localization.locale.slice(0, 2);
  } else {
    i18n.locale = lang;
  }

  AsyncStorage.setItem('language', i18n.locale).then();
});

export default i18n;
