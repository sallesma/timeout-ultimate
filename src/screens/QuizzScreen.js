import React, { useState, useLayoutEffect, useEffect, useMemo } from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, ScrollView, Text, View } from 'react-native';
import { Button } from 'react-native-elements';
import Question from '../components/Question';
import Report from '../components/Report';
import { Levels } from '../utils/config';
import I18n from '../utils/i18n';
import getQuestions from '../../data/questions';

function getRandomElementsFromArray(arr, n) {
  return arr.sort(() => 0.5 - Math.random()).slice(0, n);
}

export default (props) => {
  const { number, time, level, checkedCategories } = props.route.params;
  const [current, setCurrent] = useState(1);
  const [rightAnswersCount, setRightAnswersCount] = useState(0);
  const [canMoveForward, setCanMoveForward] = useState(false);
  const [errors, setErrors] = useState([]);
  const [showReport, setShowReport] = useState(false);
  const [questions, setQuestions] = useState([]);
  const filteredQuestions = useMemo(
    () =>
      questions.filter(
        (q) =>
          (level === Levels.ANY || level === q.level) &&
          (checkedCategories.length === 0 || checkedCategories.includes(q.category)),
      ),
    [checkedCategories, level, questions],
  );
  const selectedQuestions = useMemo(
    () => getRandomElementsFromArray(filteredQuestions, number),
    [filteredQuestions, number],
  );

  useEffect(() => {
    (async function loadResults() {
      setQuestions(await getQuestions());
    })();
  }, []);

  const onSuccess = () => {
    setRightAnswersCount(rightAnswersCount + 1);
    setCanMoveForward(true);
  };

  const onFailure = (question, checked) => {
    setCanMoveForward(true);
    setErrors(errors.concat({ question, checked }));
  };

  useLayoutEffect(() => {
    props.navigation.setOptions({
      title: `${current} / ${selectedQuestions.length}`,
      headerRight: () => (
        <Text style={styles.headerRight}>{I18n.t('quizzScreen.headerRight', { count: rightAnswersCount })}</Text>
      ),
    });
  });

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
      {selectedQuestions.length === 0 ? (
        <Text>Loading...</Text>
      ) : (
        <View>
          {!showReport && (
            <View style={styles.question}>
              <Question
                question={selectedQuestions[current - 1]}
                onSuccess={onSuccess}
                onFailure={onFailure}
                time={time}
              />
            </View>
          )}
          {canMoveForward && (
            <Button
              title={I18n.t('quizzScreen.nextQuestion')}
              onPress={() => {
                if (current === selectedQuestions.length) {
                  setShowReport(true);
                } else {
                  setCurrent(current + 1);
                }
                setCanMoveForward(false);
              }}
            />
          )}
          {showReport && (
            <Report
              rightAnswersCount={rightAnswersCount}
              quizzLength={selectedQuestions.length}
              errors={errors}
              navigation={props.navigation}
            />
          )}
        </View>
      )}
      <StatusBar style="auto" />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
  headerRight: {
    color: 'white',
    marginLeft: 16,
  },
  contentContainer: {
    paddingBottom: 64,
  },
  question: {
    marginBottom: 16,
  },
});
