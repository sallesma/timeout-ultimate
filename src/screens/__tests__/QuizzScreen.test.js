import React from 'react';
import { render } from '@testing-library/react-native';

import { Levels, Categories } from '../../utils/config';

import QuizzScreen from '../QuizzScreen';

describe('<QuizzScreen />', () => {
  it('renders correctly', async () => {
    const navigation = { setOptions: jest.fn() };
    const route = {
      params: {
        number: 10,
        time: 30,
        level: Levels.ANY,
        checkedCategories: [Categories.STATUS],
      },
    };
    const { toJSON } = render(<QuizzScreen navigation={navigation} route={route} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
