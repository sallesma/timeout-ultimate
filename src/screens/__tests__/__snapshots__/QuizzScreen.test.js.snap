// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizzScreen /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "paddingBottom": 64,
    }
  }
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Quand est-ce que le disque est dit « mort » ? (plusieurs réponses possibles)
        </Text>
        <View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#cddff3",
                "borderColor": "#ededed",
                "borderRadius": 8,
                "borderWidth": 1,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                a. Après le début d’un point, jusqu’à ce que le « pull » soit effectué
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#F4C8A2",
                "borderColor": "#ededed",
                "borderRadius": 8,
                "borderWidth": 1,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                b. Après le « pull » ou après un « turnover » quand le disque doit être amené à l’endroit où le point pivot doit être établi, jusqu’à ce qu’un pivot soit établi
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#f8f99c",
                "borderColor": "#ededed",
                "borderRadius": 8,
                "borderWidth": 1,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                c. Après un « call » qui arrête le jeu ou n’importe quel autre arrêt jusqu’à ce que le disque soit « checké »
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#e5c9e5",
                "borderColor": "#ededed",
                "borderRadius": 8,
                "borderWidth": 1,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                d. Après que le disque touche le sol jusqu’à ce que la possession soit établie par l’équipe appropriée.
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="progressbar"
          accessibilityValue={
            {
              "max": 1,
              "min": 0,
              "now": 0,
            }
          }
          accessible={true}
          onLayout={[Function]}
          style={
            [
              {
                "backgroundColor": "#f2fcfc",
                "height": 4,
                "overflow": "hidden",
                "width": "100%",
              },
              {
                "marginVertical": 10,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgb(42, 157, 157)",
                "flex": 1,
                "transform": [
                  {
                    "translateX": -0,
                  },
                  {
                    "scaleX": 0.0001,
                  },
                ],
              }
            }
          />
        </View>
        <View
          style={
            [
              {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              {
                "borderRadius": 3,
              },
              {
                "marginTop": 8,
              },
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontSize": 18,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Check the answer
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<QuizzScreen /> renders correctly 2`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "paddingBottom": 64,
    }
  }
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Quand est-ce que le disque est dit « mort » ? (plusieurs réponses possibles)
        </Text>
        <View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#cddff3",
                "borderColor": "#6aa84f",
                "borderRadius": 8,
                "borderWidth": 3,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                a. Après le début d’un point, jusqu’à ce que le « pull » soit effectué
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#F4C8A2",
                "borderColor": "#6aa84f",
                "borderRadius": 8,
                "borderWidth": 3,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                b. Après le « pull » ou après un « turnover » quand le disque doit être amené à l’endroit où le point pivot doit être établi, jusqu’à ce qu’un pivot soit établi
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#f8f99c",
                "borderColor": "#6aa84f",
                "borderRadius": 8,
                "borderWidth": 3,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                c. Après un « call » qui arrête le jeu ou n’importe quel autre arrêt jusqu’à ce que le disque soit « checké »
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#e5c9e5",
                "borderColor": "#6aa84f",
                "borderRadius": 8,
                "borderWidth": 3,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  {
                    "color": "#43484d",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                d. Après que le disque touche le sol jusqu’à ce que la possession soit établie par l’équipe appropriée.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#f2fcfc",
              "borderColor": "#2a9d9d",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#e30000",
              }
            }
          >
            Wrong answer…
          </Text>
          <Text>
            Ces 4 situations correspondent à un disque mort.
          </Text>
          <View
            style={
              [
                {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 3,
                },
                {
                  "alignSelf": "flex-end",
                  "flex": 1,
                  "marginTop": 8,
                },
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": false,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0.5,
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 0,
                    "width": 120,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#2089dc",
                      "fontSize": 14,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  See the rule
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 3,
                },
                {
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": false,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Next question
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<QuizzScreen /> renders correctly 3`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "paddingBottom": 64,
    }
  }
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <View>
      <View
        style={
          {
            "marginBottom": 8,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 26,
            }
          }
        >
          Review
        </Text>
      </View>
      <Text>
        0 correct answers out of 1 questions
      </Text>
      <Text>
        Our comment
      </Text>
      <View
        style={
          {
            "marginLeft": 16,
            "marginVertical": 8,
          }
        }
      >
        <Text>
          It's not great…
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            {
              "borderRadius": 3,
            },
            {
              "marginTop": 16,
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#2089dc",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Share my result
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            {
              "borderRadius": 3,
            },
            {
              "marginTop": 16,
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Back to the home screen
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 8,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 26,
            }
          }
        >
          Detail of the errors
        </Text>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#2a9d9d",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "marginTop": 8,
              "opacity": 1,
              "paddingVertical": 8,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "flex": 1,
                "fontSize": 14,
                "marginLeft": 8,
              }
            }
          >
            Quand est-ce que le disque est dit « mort » ? (plusieurs réponses possibles)
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<QuizzScreen /> renders correctly with no questions 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "paddingBottom": 64,
    }
  }
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <Text>
      We have no question matching your criteria.
    </Text>
  </View>
</RCTScrollView>
`;
